src/cacheModel/cacheWallet/CacheSimpleWallet.ts

```diff
@@ -0,0 +1,136 @@
+import { Address } from "../../models/account/Address";
+import { NetworkTypes } from "../../models/node/NetworkTypes";
+import { EncryptedPrivateKey } from "../../models/wallet/EncryptedPrivateKey";
+import { Password } from "../../models/wallet/Password";
+import { SimpleWallet } from "../../models/wallet/SimpleWallet";
+import {NEMLibrary} from "../../NEMLibrary";
+import { CacheAccount } from "../cacheAccount/CacheAccount";
+import { CacheAddress } from "../cacheAccount/CacheAddress";
+
+/**
+ * Cache Simple wallet model generates a private key from a PRNG
+ */
+export class CacheSimpleWallet extends SimpleWallet {
+
+  /**
+   * @internal
+   * @param name
+   * @param network
+   * @param address
+   * @param creationDate
+   * @param encryptedPrivateKey
+   */
+  constructor(name: string, network: NetworkTypes, address: CacheAddress, creationDate: LocalDateTime, encryptedPrivateKey: EncryptedPrivateKey) {
+    super(name, network, new Address(address.plain()), creationDate, encryptedPrivateKey);
+  }
+
+  /**
+   * Create a CacheSimpleWallet
+   * @param name
+   * @param password
+   * @returns {CacheSimpleWallet}
+   */
+  public static create(name: string, password: Password): CacheSimpleWallet {
+    const network = NEMLibrary.getNetworkType();
+    const wallet = nemSdk.default.model.wallet.createPRNG(name, password.value, SimpleWallet.networkTypesSDKAdapter(network));
+    return new CacheSimpleWallet(
+      name,
+      network,
+      new CacheAddress(wallet.accounts["0"].address),
+      LocalDateTime.now(),
+      new EncryptedPrivateKey(wallet.accounts["0"].encrypted, wallet.accounts["0"].iv),
+    );
+  }
+
+  /**
+   * Create a CacheSimpleWallet from private key
+   * @param name
+   * @param password
+   * @param privateKey
+   * @returns {CacheSimpleWallet}
+   */
+  public static createWithPrivateKey(name: string, password: Password, privateKey: string): CacheSimpleWallet {
+    const network = NEMLibrary.getNetworkType();
+    const wallet = nemSdk.default.model.wallet.importPrivateKey(name, password.value, privateKey, SimpleWallet.networkTypesSDKAdapter(network));
+    return new CacheSimpleWallet(
+      name,
+      network,
+      new CacheAddress(wallet.accounts["0"].address),
+      LocalDateTime.now(),
+      new EncryptedPrivateKey(wallet.accounts["0"].encrypted, wallet.accounts["0"].iv),
+    );
+  }
+
+  /**
+   * Open a wallet and generate an Account
+   * @param password
+   * @returns {CacheAccount}
+   */
+  public open(password: Password): CacheAccount {
+    const account = CacheAccount.createWithPrivateKey(this.encryptedPrivateKey.decrypt(password));
+    if (account.address.equals(this.address)) {
+      return account;
+    }
+    throw new Error("wrong password");
+  }
+
+  /**
+   * Reads the WLT content and converts it into a SimpleWallet
+   * @param {string} wlt
+   * @returns {CacheSimpleWallet}
+   */
+  public static readFromWLT(wlt: string): CacheSimpleWallet {
```

Comment:
im thinking we should keep consistent naming with the other functions that are creating a wallet

`createFromWLT`
`createFromNanoWalletWLF`

A developer would look for those names first